pkgs/build-support/trivial-builders/default.nix

```diff
@@ -802,6 +802,28 @@ rec {
         writeDirectReferencesToFile (writeText "string-file" string);
 
 
+  /**
+   * Write the references (i.e. the runtime dependencies in the Nix store) of the elements in `paths' to a file.
+   */
+  writeMultipleReferencesToFile = paths:
+    let
+      # Dump the string content / output paths of all the member of `paths` into a file
+      pathListFile = writeText "path-list" (builtins.concatStringsSep "\n"
+        # Uniquely sort to prevent unnessesary rebuilds
+        (lib.unique (builtins.sort builtins.lessThan (map toString paths)))
+      );
+    in
+    # Get the referenecs of pathListFile, but
+    # remove the line containing the path to pathListFile
+    (writeReferencesToFile pathListFile).overrideAttrs (previousAttrs: {
+      name = "multiple-runtime-deps";
+      buildCommand = previousAttrs.buildCommand + ''
+        substituteInPlace "$out" --replace ${lib.escapeShellArg pathListFile}"
+        " ""
```

Comment:
Was this linefeed at the end intentional?

Reply:
Yes. These is meant to remove the line (without leaving a blank line).

Reply:
I see. Still looks puzzling so maybe at least a comment would be appropriate. And correctness shouldn't be a concern, because `substituteInPlace` errors out when fails to match anything in the inputs

Reply:
Just modified and added comments to increase readability.